File: questaoLogica/bissexto.c
#include <stdio.h>  //incluindo a biblioteca com os metodos printf, scanf
#include <stdlib.h> //incluindo a biblioteca padrao para atribuicao e alocacao de memoria

void main()
{
    int ano, result; // definindo as variaveis do tipo int que irei usar

    do
    {
        printf("Entre com seu ano de aniversario:"); // mensagem para o usuario
        result = scanf("%d", &ano);                  // obtendo a entrada do teclado e colocando o retorno de exito ou falha do scanf em result

        if (result != 1) // se o usuario entrar com algo que nao seja um numero o result sera 0, indicando um erro
        {
            printf("Entrada invalida. Certifique-se de inserir um numero.\n");
            return; // encerro o programa
        }

        if (ano <= 0) // verifico se o usuario colocou uma data menor ou igual a zero
        {
            printf("A entrada eh invalida, ano negativo? Voce eh mais velho que cristo?\n"); // printo a mensagem de erro
        }

    } while (ano <= 0); // mantenho o laço enquanto o usuario nao colocar uma data maior que zero

    system("clear"); // limpo a janela

    if (ano < 1900) // verifico se o ano é anterior a 1900
    {
        printf("Ola pessoa do passado!\n");
    }

    if (ano > 2015 && ano < 2023) //vejo se o ano é anterior a 2023, posterior a 2015 e exibo a mensagem divertida para a criança
    {
        printf("Moshi Moshi Kodomo!");
    }

    if (ano > 2023) // verifico se o ano é posterior a 2023
    {
        printf("Ola pessoa do futuro!\n");
    }

    if (ano % 4 == 0 || (ano % 100 == 0 && ano % 400 == 0)) //verifico se o ano é bissexto
    {
        printf("O ano %d eh bissexto", ano);//printo se é bissexto
        return;//encerro o programa
    }

    printf("O ano %d nao eh bissexto", ano);//printo se não é bissexto
    return;//encerro o programa
}

File: questaoLogica/javascript/bissexto.js
function bissexto() { //criando metodo para verificar se é bissexto
    let year = document.getElementById("date").value, //crindo uma variavel year e atribuindo a ela o valor obtido no campo com id date
        message = ""; //crinado uma variavel de texto message

    if (year <= 0) { //verifico se o usuario colocou uma data menor ou igual a zero
        alert("O ano é invalido!\n Voce é mais velho que cristo?\n Coloque uma valor positivo maior que 0"); // printo a mensagem de erro
        return; //encerro a funçao
    }

    if (year < 1900) {// verifico se o ano é anterior a 1900
        message = "Ola pessoa do passado!\n"; //coloco a mensagem divertida em message
    }

    if (year < 2023 && year >= 2015) {//vejo se o ano é anterior a 2023, posterior a 2015 e exibo a mensagem divertida para a criança
        message = "Moshi Moshi Kodomo!\n"; //coloco a mensagem divertida em message
    }

    if (year > 2023) {// verifico se o ano é posterior a 2023
        message = "Ola pessoa do futuro!\n"; //coloco a mensagem divertida em message
    }

    if (year % 4 == 0 || (year % 100 == 0 && year % 400 == 0)) { //verifico se é bissexto
        alert(message + "O ano é bissexto"); //concateno as mensagens e exibo se for bissexto
        return; //encerro a função
    }

    alert(message + "O ano nao é bissexto"); //concateno as mensagens e exibo se nao for bissexto
}